crates/common/download/src/download.rs

```diff
@@ -222,7 +223,7 @@ fn create_file_and_try_pre_allocate_space(
                 file.as_raw_fd(),
                 FallocateFlags::empty(),
                 0,
-                file_len as nix::libc::off_t,
+                file_len.try_into().expect("file too large to fit in i64"),
```

Comment:
In practice it will never panic, because for the `u64 -> i64` conversion to fail, the file would have to be bigger than `2^63-1`, which comes up to roughly 8 exabytes. Maybe it is a meaningless change, but `file_len` is a function parameter, and if somebody manages to call it so horribly wrong, this would panic whereas `as` would've silently proceeded, by either truncating or overflowing, not sure which. Not that it matters because the called function would also probably return an error, but in principle, using `.into()/.try_into()` is preferred to `as`, so why not. Will probably look into converting `as` to using `into` in the rest of the codebase as well some other time.